### ISSUE TITLE
Sampled image tend to be dimmer than ddim

### ISSUE THREAD

**Issue by zxzheng826** [LuChengTHU/dpm-solver#8]
Thanks for sharing your works！ I try to apply dpm-solver to the ddim code on my own dataset. However, compared to the original ddim, the images sample from dpm is way more dimmer than those sampled from ddim. The examples are followed:
DPM：
![5](https://user-images.githubusercontent.com/76085242/194983455-149e171e-196e-42ec-b4f2-95c11d6a14aa.png)
DDIM:
![20](https://user-images.githubusercontent.com/76085242/194983611-4542a97d-b7af-4b39-8a92-d2af8850ca56.png)

The sampling procedure of DPM is indeed faster than ddim.

the code is the same as the  [example](https://github.com/LuChengTHU/dpm-solver)
`noise_schedule = NoiseScheduleVP(schedule='linear')
            model_fn = model_wrapper(
            model,
            noise_schedule,
            is_cond_classifier=False,
            total_N=1000,
            model_kwargs={}
            )
            dpm_solver = DPM_Solver(model_fn, noise_schedule)
            self.sample_dpm(dpm_solver)`

`def sample_dpm(self, dpm_solver):
        config = self.config
        total_n_samples = 50000
        img_id = len(glob.glob(f"{self.args.image_folder}/*"))
        n_rounds = (total_n_samples - img_id) // config.sampling.batch_size
       
        with torch.no_grad():
            for _ in tqdm.tqdm(
                range(n_rounds), desc="Generating image samples for FID evaluation."):
                n = config.sampling.batch_size
                x = torch.randn(
                    n,
                    config.data.channels,
                    config.data.image_size,
                    config.data.image_size,
                    device=self.device,
                )
                x_sample =dpm_solver.sample(x, steps=self.args.timesteps, eps=1e-4, adaptive_step_size=False, fast_version=True,) 
                for i in range(n):
                    tvu.save_image(x_sample[i], os.path.join(self.args.image_folder, f"{img_id}.png"))
                img_id += 1`

Could you kindly show me whether there is something wrong?

**Comment by LuChengTHU** [comment]
Hi @zxzheng826 ,

We've fixed this bug in our DPM-Solver v2.0. It is because of the error of the discrete noise schedule. We have implemented a new discrete noise schedule. Please check our new updating.

We will add more documentations and example codes in the following days.

**Comment by LuChengTHU** [comment]
Hi @zxzheng826 ,

We've updated the documentation of DPM-Solver v2.0. It seems to be very easy to use. Please check our newest version. If possible, I'd like to ask you to try it again with your project.

If you have any other questions, feel free to ask me :)

**Comment by LuChengTHU** [comment]
Hi @zxzheng826 , I've add an example of stable-diffusion at: https://github.com/LuChengTHU/dpm-solver/tree/main/example_v2/stable-diffusion  , and the converged samples between DDIM and DPM-Solver are exactly the same. Please check our new updates :)

**Comment by zxzheng826** [comment]
> Hi @zxzheng826 , I've add an example of stable-diffusion at: https://github.com/LuChengTHU/dpm-solver/tree/main/example_v2/stable-diffusion , and the converged samples between DDIM and DPM-Solver are exactly the same. Please check our new updates :)

Thank you! It's very kind of you. We will try the new code in our project once agian.

**Comment by LuChengTHU** [comment]
Hi @zxzheng826 , is there any updating?

**Comment by zxzheng826** [comment]
I found the old version code works fine in my project. Instead of using the original version of the DPM-solver noise schedule in my project, I reorganized it and utilized part of the denoising ability of DPM. 
Anyway, I appreciate your work and effort. It's a solid work! 
By the way, congratulations on the acception of NeurIPS!

Here is an example using DPM-solver in my project.  The potential of DPM and DPM-solver make it possible to reconstruct images from A to B.
![屏幕截图 2022-11-08 124311](https://user-images.githubusercontent.com/76085242/200567263-12b9e7da-12a2-44fb-95d1-46fb26bd2e38.png)
